src/main/java/com/google/devtools/build/lib/bazel/BazelRepositoryModule.java

```diff
@@ -360,6 +360,11 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
       }
 
       if (repoOptions.repositoryOverrides != null) {
+        if (pkgOptions != null && !pkgOptions.checkExternalFiles) {
+          env.getReporter()
+              .handle(Event.warn("--noexperimental_check_external_files is making it so we don't check the disk for changes to your repositories overriden with --override_repository"));
```

Comment:
Does this actually only apply to repositories overridden with override_repository, or any external repositories on disk? Sorry for not knowing the difference.

Reply:
This branch of the code (if I understand it correctly) is only ran when `--override-repository` is passed in. At least that was my intention (and I don't see the warning when using this bazel on my repository which doesn't use overrides).